### ISSUE TITLE
gitter community channel?

### ISSUE THREAD

**Issue by matanster** [apache/incubator-druid#3236]
I see druid uses mailing lists and IRC. For many projects a gitter channel works more smoothly, or as an augmentation. May I suggest considering opening an official gitter channel, if casual chat based community communication is not too much of a burden. Gitter also auto-formats code samples. (I am _not_ in any way affiliated with Gitter whatsoever, other than being a user on several channels).

**Comment by joshlemer** [comment]
Hello? No reply to this issue or to the topic in google groups for 18 months?

**Comment by joshlemer** [comment]
@gianm @leventov @jihoonson ?

**Comment by gianm** [comment]
I haven't commented so far since I don't have super strong feelings about it. Personally I like mailing lists better than chat for open source communities, since they are easier to search and work better when people operate in different time zones. But if someone wants to set one up then that is fine by me.

**Comment by joshlemer** [comment]
@gianm nobody else can make the gitter channel except people who own this repository, can you just make one it takes 30 seconds https://gitter.im/#createcommunity

**Comment by leventov** [comment]
I'm against creating a Gitter channel (or Slack, or IRC, FWIW).

**Comment by drcrallen** [comment]
There are inherent dangers to communications regarding the project that are not part of official ASF channels (i.e. the mailing list). How do other ASF projects handle such IM style communications? Are other ASF projects permissive of non-mailing-list interactions on a regular basis?

**Comment by matanster** [comment]
Dangers? your comment left me curious

**Comment by gianm** [comment]
I have hesitated to set it up myself because I am not sure if it's a good idea to have a real time chat channel. We had one in the past and it was a mixed blessing. It was great for being able to get people connected super fast, like a turbo-charged version of the mailing lists. But it was also difficult to keep the channel "staffed", if you will, since Druid committers tend to be busy people. So lots of times, someone would come in looking for help and there would be nobody there to help them. That side of things ended up being frustrating for everyone, compared to the mailing lists, where it was more likely that someone would chime in with thoughts when they could.

I probably wouldn't personally a chat channel, because if I joined it, then it would make me feel bad when someone asks for help and I'm not available to help them. And I feel that a channel should be created by someone who is also volunteering to join and monitor it.

@leventov could you elaborate more on why you're against a channel?

@drcrallen my understanding of ASF policy is that decisions must be recorded on ASF infra (i.e. mailing lists or wiki), for legal and recordkeeping reasons. (Btw: github counts, since we archive all github activity to commits@druid.apache.org.) But I assume that this is less strict for general, non-decision-making communication. Fwiw, Kafka has an IRC channel (https://kafka.apache.org/contact). It seems to get a couple of messages a day.

**Comment by leventov** [comment]
@gianm I'm against a Gitter channel for the same reasons as you mentioned, plus I dislike any chats for several more reasons (chats are time sinks; promote shallow thinking; unstructured, it's hard to follow conversations there; not indexed by search engines; etc.)

There are plenty of mediums where somebody could ask questions: Github issues, mailing list, [StackOverflow](https://stackoverflow.com/questions/tagged/druid).

**Comment by matanster** [comment]
Just to note, chat on gitter/slack work extremely well for some technical communities, some of your aversion is not based on facts but excessive personal apprehension.

**Comment by gianm** [comment]
We do already have an IRC channel, and there are some nice web based IRC clients. IMO if you are interested in a chat channel being a more active part of the community, the best way to accomplish that is to take charge: join the IRC channel, encourage others to do so in the easiest way possible, and help people when they show up asking about stuff. If enough people do that then it will become a thing.

**Comment by joshlemer** [comment]
@gianm @leventov just because you don't personally like to use these tools doesn't mean that others don't find them useful. I personally participate in many gitter chats including those for the Scala and Rust communities, and including apache projects like Beam and Spark (Beam has a slack channel on the official apache slack instance, and spark has many gitter channels). While many questions that go on in these channels are beginner questions, that's fine. And at the same time, lots of very senior people in the community hang out in there and talk about very advanced topics. 

IRC is terrible for online discussion because you can't see history, you can't share code, you can't see who has seen your messages, there are no avatars / connection to identity, etc etc etc. At least consider the fact that multiple people have asked for it, which implies a much larger amount of people who would use it but haven't asked.

**Comment by asdf2014** [comment]
Hi, all. IMHO, we all know the shortcomings of live chat, we can't archive these chats as a structured document system for later search. However, it still cannot deny its advantages. We can use timely communication tools to bring together Druid users and developers, making communication and thought collisions more convenient. In china, we created some druid QQ / wechat groups to talk, and we helped each other solve many problems, also we shared our thoughts and resources in group. If you feel that it is necessary, I can contribute some QQ, wechat groups. However, this still depends on you. BTW, [PR#6002](https://github.com/apache/incubator-druid/pull/6002) has been created for IRC#druid-dev,  and we will see [![IRC#druid-dev](https://img.shields.io/badge/IRC-druid--dev-blue.svg)](https://webchat.freenode.net?channels=druid-dev) in README doc. In addition, we can add qq groups like this. [![1670647](https://img.shields.io/badge/QQ%E7%BE%A4-1670647-blue.svg)](https://shang.qq.com/wpa/qunwpa?idkey=f86b3c8de20da1658a3bb42df17a2fc4eee0d75c4a130a63585fdd257e3565ed)

**Comment by gianm** [comment]
FYI, it turns out the official ASF Slack team ([the-asf.slack.com](https://the-asf.slack.com)) has a **#druid** channel. There's now a signup link on https://druid.apache.org/community/ (check the "Slack" line).

> I probably wouldn't personally join a chat channel, because if I joined it, then it would make me feel bad when someone asks for help and I'm not available to help them. And I feel that a channel should be created by someone who is also volunteering to join and monitor it.

I changed my mind on this, btw, and I've joined the Slack channel above. Hope to see anyone there that is interested in chatting.

Let's keep the discussions there to user questions, since dev discussions should happen on the dev@druid.apache.org.
